parsec/core/persistent_storage.py

```diff
@@ -211,48 +218,49 @@ def _is_block(self, conn: Connection, path: Path, block_id: BlockID):
 
     def _get_block(self, conn: Connection, path: Path, block_id: BlockID):
         with self._open_cursor(conn) as cursor:
+            cursor.execute("BEGIN")
             cursor.execute(
-                """UPDATE blocks SET accessed_on = ? WHERE block_id = ?""",
-                (str(pendulum.now()), str(block_id)),
+                """
+                UPDATE blocks SET accessed_on = ? WHERE block_id = ?;
+                """,
+                (time(), str(block_id)),
             )
             cursor.execute("SELECT changes()")
             changes, = cursor.fetchone()
+            if not changes:
+                raise LocalStorageMissingError(block_id)
 
-        if not changes:
-            raise LocalStorageMissingError(block_id)
+            cursor.execute("""SELECT data FROM blocks WHERE block_id = ?""", (str(block_id),))
+            ciphered, = cursor.fetchone()
+            cursor.execute("END")
 
-        ciphered = self._read_file(block_id, path)
         return decrypt_raw_with_secret_key(self.local_symkey, ciphered)
 
     def _set_block(self, conn: Connection, path: Path, block_id: BlockID, raw: bytes):
```

Comment:
A few things need to be removed:
- the `path` argument for `_set_block` and `_get_block`
- the `_dirty_db_files` and `_clean_db_files` attributes

Reply:
done \o/
(also removed `get_block_cache_size` which is meaningless and wasn't used anyway)